src/RelayClient.ts

```diff
@@ -624,18 +674,22 @@ class RelayClient extends EnvelopingEventEmitter {
   }: EnvelopingTxRequest): Promise<void> {
     const { signature } = metadata;
     const {
+      request: { tokenContract },
       relayData: { callVerifier },
     } = relayRequest;
 
     const provider = getProvider();
 
-    if (isDeployRequest(relayRequest)) {
+    if (
```

Comment:
Just to clarify, at the moment if it's a deploy and we expect the payment in native, we don't do anything. Is that correct?

Reply:
right, this is wrong, let me remove it since now we need to validate it with the verifier that was developed.

Reply:
updated